src/renderer/ScreenController.tsx

```diff
@@ -354,19 +337,33 @@ export default class ScreenController extends Component {
             screen: this.state.screen,
           }}
         >
-          <KeybindingsContextProvider>
-            <div className='main-container'>
-              <AccountListSidebar
-                selectedAccountId={this.selectedAccountId}
-                onAddAccount={this.addAndSelectAccount.bind(this)}
-                onSelectAccount={this.selectAccount.bind(this)}
-                openAccountDeletionScreen={this.openAccountDeletionScreen.bind(
-                  this
-                )}
-              />
-              {this.renderScreen()}
-            </div>
-          </KeybindingsContextProvider>
+          {/*
+            The key attribute here forces a clean re-rendering when the
+            account changes. We needs this to reset the chat context state.
+          */}
+          <ChatProvider
+            key={this.selectedAccountId}
+            accountId={this.selectedAccountId}
+          >
+            <ContextMenuProvider>
+              <DialogContextProvider>
+                <RuntimeAdapter />
+                <KeybindingsContextProvider>
+                  <div className='main-container'>
+                    <AccountListSidebar
+                      selectedAccountId={this.selectedAccountId}
+                      onAddAccount={this.addAndSelectAccount.bind(this)}
+                      onSelectAccount={this.selectAccount.bind(this)}
+                      openAccountDeletionScreen={this.openAccountDeletionScreen.bind(
+                        this
+                      )}
+                    />
+                    {this.renderScreen()}
+                  </div>
+                </KeybindingsContextProvider>
+              </DialogContextProvider>
+            </ContextMenuProvider>
+          </ChatProvider>
```

Comment:
This became a little cute context sandwich. Many dialogues are dependent on knowing about the chat state, while every screen needs to know about dialogues, this is why `ChatProvider` pops up in a fairly "global" position.

I don't think this is bad at all (a currently selected chat _is_ a very "global" state in this application), just wanted to note that here.